flashinfer/gemm.py

```diff
@@ -1883,22 +1903,30 @@ def mm_fp4(
             f"a_descale and b_descale must have float8_e4m3fnx2 packed into uint8. "
             f"Got {a_descale.dtype} and {b_descale.dtype}."
         )
-    if alpha.dtype != torch.float:
+    if alpha is not None and alpha.dtype != torch.float:
         raise ValueError(f"alpha must be a float tensor, got {alpha.dtype}")
-    if alpha.numel() != 1:
+    if alpha is not None and alpha.numel() != 1:
         raise ValueError(f"alpha must be a scalar, got {alpha.numel()}")
 
+    if alpha is None and use_nvfp4:
+        raise ValueError("alpha must be provided for nvfp4 quantization.")
+
     if out_dtype not in (torch.bfloat16, torch.float16):
         raise ValueError(
             f"Unsupported output dtype: {out_dtype}. "
             f"Only torch.bfloat16 and torch.float16 are supported for FP4 GEMM operations."
         )
-    if block_size != 16:
-        raise ValueError("Only block_size = 16 is supported for FP4 GEMM operations.")
+
+    if use_nvfp4 and block_size != 16:
+        raise ValueError("nvfp4 only supports block_size = 16.")
+    if not use_nvfp4 and block_size not in [16, 32]:
```

Comment:
should we only allow block size 32 for mxfp4?
I checked some reference: 
https://docs.vllm.ai/en/latest/api/vllm/model_executor/layers/quantization/mxfp4.html#__codelineno-0-279
https://developer.nvidia.com/blog/nvfp4-trains-with-precision-of-16-bit-and-speed-and-efficiency-of-4-bit/